source/ui.js

```diff
@@ -119,29 +121,22 @@ const checkNewFilesAndDependencies = async (pkg, rootDir) => {
 	return answers.confirm;
 };
 
-// eslint-disable-next-line complexity
-const ui = async (options, {pkg, rootDir, isYarnBerry = false}) => {
+/**
+ * @param {typeof import('./cli-implementation.js').cli['flags']} options
+ * @param {{pkg: import('read-pkg').NormalizedPackageJson; rootDir: string}} context
+ */
```

Comment:
```suggestion
/**
@param {typeof import('./cli-implementation.js').cli['flags']} options
@param {{pkg: import('read-pkg').NormalizedPackageJson; rootDir: string}} context
*/
```

Reply:
Both [jsdoc](https://jsdoc.app/tags-param) and [tsdoc](https://tsdoc.org/pages/tags/param/) use the form I have here (as well as VSCode's code-inserter feature). I don't feel strongly though, so feel free to commit that suggestion if you want the no-`*` form anyway.

Reply:
I see that other jsdoc are using the format you suggested. I still think it's better to align with conventions but no need to introduce inconsistency with this PR. I changed all my new ones to match what's in the repo.

Reply:
Related to my other comment - this could then be:

```js
@param {import('./cli-implementation.js').CLI['flags']} options
```